### DISCUSSION TITLE
React Native navigation across the board?

### DISCUSSION THREAD

**Discussion by ghost** [mmazzarolo/react-native-universal-monorepo#7] [Q&A]
Hey, thanks for setting this up. It's great to see a template like this that enables us to quickly get going with multi platform support beyond just Android and iOS. I'm quite keen to use React Native for web for the first time after years of writing React native and React web apps separately sharing business logic only.

Quick question, from the project setup, the whole idea is to write a single React Native app that works across the board e.g android, ios and web and so on.

For this, I wonder how navigation would work or whether you could share your strategy around this. Could you share your experience in this area? 

What I'm looking at the moment is if I setup all the navigation in core folder using react-navigation/native whether that would work on web too. I'm more reading as well.

**Comment by mmazzarolo** [comment]
Hey @proverbface !  Thanks!

> the whole idea is to write a single React Native app that works across the board e.g android, ios and web and so on. 

Kinda? I'd say the goal here is to lay a foundation for enabling it, but let's keep in mind that (as of today) it's just a subset of React Native libraries that support multiple platforms.  
So pros/cons of this approach must still be evaluated before adopting it.  
The good news is that the monorepo setup here is still valuable even if you're planning to just share core business JavaScript logic between a React Native and a React app 👍 

> For this, I wonder how navigation would work or whether you could share your strategy around this. Could you share your experience in this area?
> What I'm looking at the moment is if I setup all the navigation in core folder using react-navigation/native whether that would work on web too. I'm more reading as well.

[React Navigation works on the web](https://reactnavigation.org/docs/web-support/), but it's still in a WIP state. I've been able to use it in a tiny side project with Create React App, but I wasn't doing anything fancy with it.  

I think that something valuable for the repo could be to collect in the README.md these discussions and notes about the integration of common libraries across multiple platforms. Either by linking the GitHub discussion or by adding the notes directly in the README.  

What do you think?

**Comment by ericlifs** [comment]
Just my two cents here, I'm working on a similar monorepo setup (currently we are supporting web, iOS and tvOS) and what we actually did was to create a mobx store (name any state management library you want) for managing the navigation status and then each platform app can respond accordingly as each platform knows (web uses react router and iOS/tvOS uses react-navigation).

For now, as @mmazzarolo mentions the implementation of React Navigation on the web is kinda short so that's why we decided to take that former approach I was talking about. I have not played with React Navigation on the web honestly, but another cause of going with this approach is because usually things tend to work better when they are closer to how things are natively implemented. 

We can connect if you want and I can explain a little bit more the details of our implementation.
